src/util/device.rs

```diff
@@ -37,8 +39,10 @@ pub trait DeviceExt {
 impl DeviceExt for crate::Device {
     fn create_buffer_init(&self, descriptor: &BufferInitDescriptor<'_>) -> crate::Buffer {
         let unpadded_size = descriptor.contents.len() as crate::BufferAddress;
-        let padding = crate::COPY_BUFFER_ALIGNMENT - unpadded_size % crate::COPY_BUFFER_ALIGNMENT;
-        let padded_size = padding + unpadded_size;
+        // Valid vulkan usage is
+        // 1. buffer size must be a multiple of COPY_BUFFER_ALIGNMENT.
+        // 2. buffer size must be greater than 0.
+        let padded_size = ((unpadded_size + COPY_BUFFER_ALIGNMENT - 1) / COPY_BUFFER_ALIGNMENT).max(1) * COPY_BUFFER_ALIGNMENT;
```

Comment:
We can be a smidge cleverer here:
```suggestion
        let align_mask = COPY_BUFFER_ALIGNMENT - 1;
        let padded_size = ((unpadded_size + align_mask) & align_max).max(1);
```
But otherwise looks good!

Reply:
Hmm, yes if we assume the alignment is a power of two (which I guess it will always be) then that will work. Thanks!
Though your `max` call should be `max(COPY_BUFFER_ALIGNMENT)` in this case. I'll make that change separately.

Reply:
Actually I checked the assembly, and llvm generates the exact same output for both cases. Should I still change this? I suppose it just comes down to what is more readable.

Reply:
Ah yes, good catch :)

Reply:
IMO it's clearer this way as the whole divide then multiply thing always takes me a minute to properly parse what's going on.